plugins/org.jboss.tools.cdi.reddeer/src/org/jboss/tools/cdi/reddeer/validators/AbstractValidationProvider.java

```diff
@@ -40,4 +49,31 @@ public ValidationProblem getValidationProblem(ValidationType type){
 		return null;
 	}
 	
+	protected void decoratorValidationProviderInit() {
```

Comment:
I definitely would not place decorator's validation messages in the abstract class that all other validators are derived from. How about that you have one DecoratorValidationProvider and it has one class member - jsr which is defined when class is instantiated. In such case this could be applied for any CDI specification. There would not be even a need for CDIX validation provider class for each CDI specification (test class). Of course, if the problems are the same for all CDI specs.

Application: 
```
TestClassForDecoratorValidationCDI20 {
   @Before
   public void before() {
      validationProvider = new DecoratorValidtionProvider("JSR-365");   
   }
}
```

Reply:
Fixed.